client/web/src/components/externalServices/AddExternalServicesPage.tsx

```diff
@@ -53,10 +54,27 @@ export const AddExternalServicesPage: FC<AddExternalServicesPageProps> = ({
         setHasDismissedPrivacyWarning(true)
     }
 
-    const id = new URLSearchParams(search).get('id')
+    const params = new URLSearchParams(search)
+    const id = params.get('id')
     if (id) {
         const externalService = allExternalServices[id]
         if (externalService) {
+            if (externalService.kind === ExternalServiceKind.GITHUB) {
+                const appID = params.get('appID')
+                const installationID = params.get('installationID')
+                const baseURL = params.get('url')
+                const org = params.get('org')
+                if (appID !== null && installationID !== null && baseURL !== null) {
```

Comment:
I think we need to protect against `undefined` or empty strings in here as well. I think this would do it (and also looks much nicer):
```suggestion
                if (appID && installationID && baseURL) {
```